app/javascript/elm/tests/SensorTests.elm

```diff
@@ -145,38 +129,34 @@ all =
                         )
         , test "nameForSensorId returns the name given the sensorId" <|
             \_ ->
-                [ { sensor
-                    | id_ = "id"
-                    , name = "name"
-                  }
-                , { sensor
-                    | id_ = "other id"
-                    , name = "other name"
-                  }
-                ]
-                    |> nameForSensorId "id"
+                let
+                    sensor1 =
+                        { sensor | name = "name" }
+
+                    sensor2 =
+                        { sensor | name = "other name" }
+                in
+                [ sensor1, sensor2 ]
+                    |> nameForSensorId (toId sensor1)
```

Comment:
hm, I feel like when I'm reading such a test description I'd expect the test to call `nameForSensorId` on a `sensorId` instead of going deeper and calling `toId`. then `toId` and `toLabel` could be tested separately. I guess this is a more "integration-like" approach? in that case, since the input is actually a `sensor` not a `sensorId`, the test description seems off. I'm probably nitpicking.

Reply:
I see what you mean but in this case it's much more convenient to "integrate". Otherwise I would have to write a correct id by hand or copy the logic from the production code which seems smelly